library/std/src/sys/unix/os.rs

```diff
@@ -543,7 +543,21 @@ pub fn getenv(k: &OsStr) -> Option<OsString> {
     }
 }
 
+fn process_is_threaded() -> bool {
+    if cfg!(any(target_os = "linux", target_os = "android")) {
+        if let Ok(r) = crate::fs::read_dir("/proc/self/task") {
```

Comment:
Rsix checks that `/proc` is an actual proc fs: https://github.com/bytecodealliance/rustix/blob/3e20a74673b1faa3d2c9bf7f4182a68173ef3532/src/io/procfs.rs#L199-L206

Reply:
OK yeah...the rustix code is really nice I have to say.

But, we'll cross the "serious implementation of `process_is_threaded()`" bridge when(if?) there is agreement to do a PR adding `try_set_env()` to the std.